tools/codesign_mac_executable.py

```diff
@@ -0,0 +1,232 @@
+import shutil
+import subprocess
+import time
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--macos_executable", required=True)
+    parser.add_argument("--entitlements_file", required=True)
+    parser.add_argument("--developer_id", required=True)
+    parser.add_argument("--output_dir", required=True)
+    parser.add_argument("--apple_username", required=True)
+    parser.add_argument("--apple_password", required=True)
+    parser.add_argument("--wait_time_seconds", type=float, default=600.0)
+    args = parser.parse_args()
+
+    assert os.path.exists(args.macos_executable)
+    radio_effect_units = [
+        os.path.join(args.macos_executable, "Contents", "Resources", "RadioEffectUnit.component"),
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Build", "Mac", "RadioEffectUnit")]
+
+    print(f"[SPEAR | codesign_mac_executable.py] Removing {radio_effect_units[0]}.")
+    shutil.rmtree(radio_effect_units[0], ignore_errors=True)
+
+    print(f"[SPEAR | codesign_mac_executable.py] Removing {radio_effect_units[1]}.")
+    shutil.rmtree(radio_effect_units[1], ignore_errors=True)
+
+    print(f"[SPEAR | codesign_mac_executable.py] Changing rpaths...")
+
+    # change current working directory to add relative rpaths 
+    cwd = os.getcwd()
+    os.chdir(os.path.join(args.macos_executable, ".."))
+
+    executable_name = os.path.basename(args.macos_executable)
+
+    cmd = [
+        "install_name_tool",
+        "-add_rpath",
+        os.path.join(executable_name, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "Ogg", "Mac"),
+        os.path.join(executable_name, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "Ogg", "Mac", "libogg.dylib")]
+    print(f"[SPEAR | codesign_mac_executable.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd)
+    assert cmd_result.returncode == 0
+
+    cmd = [
+        "install_name_tool",
+        "-add_rpath",
+        os.path.join(executable_name, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "Vorbis", "Mac"),
+        os.path.join(executable_name, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "Vorbis", "Mac", "libvorbis.dylib")]
+    print(f"[SPEAR | codesign_mac_executable.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd)
+    assert cmd_result.returncode == 0
+
+    cmd = [
+        "install_name_tool",
+        "-add_rpath",
+        os.path.join(executable_name, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac"),
+        os.path.join(executable_name, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libPxFoundation.dylib")]
+    print(f"[SPEAR | codesign_mac_executable.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd)
+    assert cmd_result.returncode == 0
+
+    cmd = [
+        "install_name_tool",
+        "-rpath",
+        "/Volumes/Work/Perforce/UE4/Engine/Binaries/ThirdParty/PhysX3/Mac",
```

Comment:
Is this absolute path going to cause problems for me when I try to execute this script?

Reply:
We are removing this absolute path from the dylibs.

Reply:
Makes sense.